pkg/icingadb/v1/history/notification.go

```diff
@@ -42,10 +44,49 @@ func (*HistoryNotification) TableName() string {
 	return "history"
 }
 
+// NotificationType represents the type of notification for a notification history entry.
+//
+// Starting with Icinga 2 v2.15, the type is will always be written to Redis as a string.
+// This merely exists to provide a compatibility with older history entries lying around in Redis,
+// which may have been written as an integer.
+type NotificationType string
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface.
+func (nt *NotificationType) UnmarshalText(text []byte) error {
+	t := string(text)
+	for bitset, name := range notificationTypes {
+		if t == name || t == bitset {
+			*nt = NotificationType(name)
+			return nil
+		}
+	}
+
+	return errors.Errorf("bad notification type: %#v", t)
+}
```

Comment:
How much do we care for actually detecting invalid values here? I mean we would have been fine with this being just a plain `string` that would do no validation, this type exists only to provide compatibility.

That would then allow simplifying this a bit (and could also be used for the other two types where it would avoid spelling out each string literal twiche):

```suggestion
func (nt *NotificationType) UnmarshalText(text []byte) error {
	switch t := string(text); t {
	case "1":
		*nt = "downtime_start"
	case "2":
	// [...] (only cases for numbers)
	default:
		*nt = t
	}
	return nil
}
```